### ISSUE TITLE
[BUG] Recursive schema on array field causes stack overflow

### ISSUE THREAD

**Issue by ajwootto** [DouglasGabr/mongoose-lean-defaults#25]
**Describe the bug**  
If you define a schema that has a recursive reference to itself (using Schema.add) and that recursive reference occurs inside an array field on the schema, the plugin will trigger a stack overflow exception when trying to query with lean defaults.

**To Reproduce**  
```javascript
import { Schema, model, connect, connection } from 'mongoose'
import leanDefaults from 'mongoose-lean-defaults'

const MySchema = new Schema({
    name: String
})

const Test = new Schema({ subField: { type: [MySchema] } })
MySchema.add(Test)

MySchema.plugin(leanDefaults)

const MyModel = model('MyModel', MySchema)

async function execute() {

    await connect('mongodb://localhost:27017/')
    await connection.dropDatabase()

    const myModel = new MyModel({
        name: 'test',
        subField: [{
            subField: [{
                name: 'test'
            }]
        }]
    })

    await myModel.save()

    const result1 = await MyModel.findOne({}).lean().exec()
    console.log(result1)

    // This will throw a stack overflow error
    await MyModel.findOne({}).lean({ defaults: true }).exec()
}

execute()

```

Results in:
```
(node:7741) UnhandledPromiseRejectionWarning: RangeError: Maximum call stack size exceeded
    at RegExp.test (<anonymous>)
    at Object.exports.get (node_modules/mpath/lib/index.js:71:40)
    at node_modules/mpath/lib/index.js:79:21
    at Array.map (<anonymous>)
    at Object.exports.get (node_modules/mpath/lib/index.js:77:29)
    at node_modules/mpath/lib/index.js:79:21
    at Array.map (<anonymous>)
    at Object.exports.get (node_modules/mpath/lib/index.js:77:29)
    at model.Query.attachDefaults (node_modules/mongoose-lean-defaults/index.js:86:42)
    at model.Query.attachDefaults (node_modules/mongoose-lean-defaults/index.js:90:28)
```

**Expected behavior**  
I know this is a pretty big edge case, but we need to represent data this way for our use case. The actual schema definition seems to work fine for all other mongoose operations, it's only when this plugin is applied that we get the `call stack size exceeded` error.

**Additional context**  
Mongoose version:  6.0.11
Node version: 14.15

**Comment by DouglasGabr** [comment]
Sorry for the late response. Fixed in 2.1.0

**Comment by ajwootto** [comment]
Thanks @DouglasGabr !
